internal/config/ociartifact/ociartifact.go

```diff
@@ -164,6 +143,130 @@ func (o *OCIArtifact) Pull(ctx context.Context, img string, opts *PullOptions) (
 	}, nil
 }
 
+func (o *OCIArtifact) GetManifest(ctx context.Context, img string, opts *PullOptions) (manifest.Manifest, error) {
+	log.Infof(ctx, "Pulling manifest from ref: %s", img)
+
+	// Use default pull options
+	if opts == nil {
+		opts = &PullOptions{}
+	}
+
+	src, err := o.getSourceFromImageName(ctx, img, opts)
+	if err != nil {
+		return nil, err
+	}
+
+	parsedManifest, err := o.getParsedManifest(ctx, src, opts)
+	if err != nil {
+		return nil, err
+	}
+
+	return parsedManifest, nil
+}
+
+func (o *OCIArtifact) GetConfig(ctx context.Context, img string, opts *PullOptions) (*v1.Image, error) {
```

Comment:
_In general_, artifacts don’t have image configs. If they have image configs, they are images. (I can see a semantic argument that “image is an artifact”, but… what are we doing here?)